airflow/models.py

```diff
@@ -2662,10 +2662,15 @@ def __init__(
         }
 
     def __eq__(self, other):
-        return (
-            type(self) == type(other) and
-            all(self.__dict__.get(c, None) == other.__dict__.get(c, None)
-                for c in self._comps))
+        if (type(self) == type(other) and
+                self.task_id == other.task_id):
```

Comment:
Do we need to also check that `self.dag_id == other.dag_id`?

Reply:
(And if we do we should probably add a test that checks that covers that)

Reply:
I think this is implicitly well covered by the cycle detection tests in `tests/models.py`.

Reply:
```
dag1 = DAG('dag2')
task1 = MyOp(task_id='start', dag=dag1)

dag2 = DAG('dag1')
task2 = MyOp(task_id='start', dag=dag2)
```

Will `task1 == task2`? It sort of looks like form this diff only that it might.

Reply:
So comparing the `task_id`s is just an optimization. If they differ, the attributes `task_id`, `dag_id`, `owner`, `email`, `email_on_retry`, `retry_delay`, `retry_exponential_backoff`, `max_retry_delay`, `start_date`, `schedule_interval`, `depends_on_past`, `wait_for_downstream`, `adhoc`, `priority_weight`, `sla`, `execution_timeout`, `on_failure_callback`, `on_success_callback` and `on_retry_callback` are compared to determine if the dags are really equal.

Reply:
Oh sorry yes! If they don't match then we return false. Right